packages/core/src/utils/CoreHelperUtil.ts

```diff
@@ -267,5 +267,9 @@ export const CoreHelperUtil = {
     }
 
     return false
+  },
+
+  uniqueBy<T>(arr: T[], key: keyof T) {
+    return Array.from(new Map(arr.map(i => [i[key], i])).values())
```

Comment:
@tomiir 

Seems we have many iterations here so I wanted to check O(n) complexity. 

While the current solution have the O(n) complexity, I do think we could make it more performant:
- `arr.map(I => [I[key], I])`: O(n)
- `new Map(arr.map(...))`: O(n)
- `Array.from(map.values())`: O(n)

O(n) + O(n) + O(n) = O(3n), which is equals to O(n).

If we go with the following approach below, it's still O(n) but since the only iteration is `.filter`, the general performance is much faster. 

```ts
function uniqueBySet<T>(arr: T[], key: keyof T): T[] {
  const seen = new Set();
  return arr.filter(item => {
    const keyValue = item[key];
    if (seen.has(keyValue)) {
      return false;
    }
    seen.add(keyValue);
    return true;
  });
}
```

Compared both functions with `10_000` items here is the results:

- uniqueByMap took `2.899999976158142` ms
- uniqueBySet took `0.699999988079071` ms

with `100_000` items:
- uniqueByMap took `13.800000011920929` ms
- uniqueBySet took `2.2999999821186066` ms

Reply:
updated